src/MinterGateway.sol

```diff
@@ -602,10 +602,70 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
     }
 
     /// @inheritdoc IMinterGateway
-    function getPenaltyForMissedCollateralUpdates(address minter_) external view returns (uint240) {
-        uint112 penaltyPrincipal_ = _getPenaltyPrincipalForMissedCollateralUpdates(minter_);
+    function getPenalties(
+        address minter_
+    ) external view returns (uint240 missedIntervalPenalty_, uint240 undercollateralizedPenalty_) {
+        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
+
+        if (principalOfActiveOwedM_ == 0) return (0, 0);
+
+        uint32 penaltyRate_ = penaltyRate();
+
+        if (penaltyRate_ == 0) return (0, 0);
```

Comment:
We could combine the conditions.
```suggestion
        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
        uint32 penaltyRate_ = penaltyRate();
        
        if (principalOfActiveOwedM_ == 0 || penaltyRate_ == 0) return (0, 0);
```

Reply:
Less gas efficient that returning early without an additional couple of SLOADs and calls if `principalOfActiveOwedM_ == 0`.